campaignion_email_to_target/src/Wizard/TargetStep.php

```diff
@@ -88,8 +88,11 @@ public function stepForm($form, &$form_state) {
     $settings['validations'] = [
       // Escape backslashes so JS won’t interpret them as escape sequence.
       'email' => '^[^\\s@]+@[^\\s@]+\\.[^\\s@]+$',
-      'first_name' => '\\S+',
-      'last_name' => '\\S+',
+      // First and last name are:
+      // - at least one non-whitespace char (which is not ,"@ too)
+      // - and no ,"@ chars in the whole string.
+      'first_name' => '^[^,"@]*[^,"@\\s][^,"@]*$',
+      'last_name' => '^[^,"@]*[^,"@\\s][^,"@]*$',
```

Comment:
Would `'^\\s*([^,"@\\s]\\s*)+$'` be more readable? I would read that as whitespace followed by (at least one) whitespace separated words (with blacklisted chars).

Reply:
yes, probably more readable. (it's not a trivial Regexp anyways.) will change.